private/pkg/slogtest/slogtest.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package slogtest provides Loggers for testing.
+package slogtest
```

Comment:
Suggestion, we name this literally anything else, this conflicts with https://pkg.go.dev/testing/slogtest which trips up imports.

Maybe we could have just an `slogext` package as the one way within buf to extend (and work) with slog, e.g., 

slogext.NewTestLogger
slogext.NopLogger
slogext.NewBuilder

etc.

Reply:
I'll rename to `slogtestext` for now. I have this long-term plan to rename all the `.*ext` stuff to `x.*` ie `xslices`, `xslog`, but haven't gotten there yet. By naming this `slogtestext`, it'll map to `xslogtest` in that world.

Reply:
Done